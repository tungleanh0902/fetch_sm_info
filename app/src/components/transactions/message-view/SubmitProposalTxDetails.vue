<template>
  <div class="tx__content">
    <TransactionIcon :transaction-type="type" />
    <div class="tx__content__left">
      <h3>{{ caption }}</h3>
      <span>{{ transaction.details.proposalTitle }}</span>
    </div>
    <div class="tx__content__right">
      <p class="amount">
        {{ transaction.details.initialDeposit.amount | prettyLong }}&nbsp;
        {{ transaction.details.initialDeposit.denom }}
      </p>
    </div>
  </div>
</template>

<script>
import { prettyLong } from "scripts/num.js"
import TransactionIcon from "../TransactionIcon"

export default {
  name: `submit-proposal-tx-details`,
  filters: {
    prettyLong,
  },
  components: {
    TransactionIcon,
  },
  props: {
    transaction: {
      type: Object,
      required: true,
    },
    validators: {
      type: Object,
      required: true,
    },
  },
  data: () => {
    return {
      type: `Submitted`,
      caption: `Submitted proposal`,
    }
  },
}
</script>
