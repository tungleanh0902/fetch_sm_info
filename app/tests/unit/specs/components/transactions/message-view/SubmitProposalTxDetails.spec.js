import { shallowMount } from "@vue/test-utils"
import SubmitProposalTxDetails from "src/components/transactions/message-view/SubmitProposalTxDetails"

describe(`SubmitProposalTxDetails`, () => {
  let wrapper

  const tx = {
    type: "SubmitProposalTx",
    hash: "47E70B788A3DEA7D7244A4EC0FC58B4FEAA36456541F94AD30BF37D0C6D96C1D",
    height: 779937,
    timestamp: "2020-02-10T10:13:15Z",
    memo: "",
    success: true,
    fees: [],
    details: {
      proposalType: "cosmos-sdk/TextProposal",
      proposalTitle: "Lunie was here!",
      proposalDescription: "Blah blah blah and blah blah blah ...",
      initialDeposit: {
        denom: "ATOM",
        amount: "10",
      },
    },
  }

  it(`renders a submit proposal transaction message`, () => {
    wrapper = shallowMount(SubmitProposalTxDetails, {
      propsData: {
        transaction: tx,
        validators: {},
      },
      stubs: [`router-link`],
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
