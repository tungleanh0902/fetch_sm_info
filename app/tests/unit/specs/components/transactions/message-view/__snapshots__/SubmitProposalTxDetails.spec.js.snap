// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitProposalTxDetails renders a submit proposal transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiontype="Submitted"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Submitted proposal
    </h3>
     
    <span>
      Lunie was here!
    </span>
  </div>
   
  <div
    class="tx__content__right"
  >
    <p
      class="amount"
    >
      
      10Â 
      ATOM
    
    </p>
  </div>
</div>
`;
